servers/src/grin/sync/body_sync.rs

```diff
@@ -142,6 +144,17 @@ impl BodySync {
 			.collect::<Vec<_>>();
 
 		if hashes_to_get.len() > 0 {
+			// Stuck detection:
+			// In case stuck detected, exit sync thread to avoid useless infinite loop.
+			if hashes_to_get.len() == 1 && hashes_to_get[0].clone() == self.stuck_detector.0 {
+				self.stuck_detector.1 += 1;
+				if self.stuck_detector.1 >= 100 {
+					panic!("sync: body sync stuck detected! sync thread exited!");
```

Comment:
sorry I'm out of the loop, what does panic! do? Kill the current thread? Will it get resurrected, and if yes, where would that thread usually be killed if no stuck detected?

Reply:
`panic!` here is used here to exit the sync thread.
no, this thread will not get resurrected.